public/docs/_examples/toh-5/dart/lib/hero_detail_component.dart

```diff
@@ -0,0 +1,56 @@
+// #docplaster
+// #docregion
+// #docregion v2
+// #docregion import-oninit
+import 'package:angular2/core.dart';
+// #enddocregion import-oninit
+// #docregion import-route-params
+import 'package:angular2/router.dart';
+// #enddocregion import-route-params
+
+import 'hero.dart';
+// #docregion import-hero-service
+import 'hero_service.dart';
+// #enddocregion import-hero-service
+import 'dart:html';
+
+// #docregion extract-template
+@Component(
+  selector: 'my-hero-detail',
+  // #docregion template-url
+  templateUrl: 'hero_detail_component.html',
+  // #enddocregion template-url
+// #enddocregion v2
+  styleUrls: const ['hero_detail_component.css']
+// #docregion v2
+)
+// #enddocregion extract-template
+// #docregion implement
+class HeroDetailComponent implements OnInit {
+// #enddocregion implement
+// #docregion ctor
+  HeroDetailComponent(this._heroService, this._routeParams);
+
+  HeroService _heroService;
+
+  RouteParams _routeParams;
+// #enddocregion ctor
+  Hero hero;
+
+// #docregion ng-oninit
+  ngOnInit() async {
+    // #docregion get-id
+    var id = int.parse(_routeParams.get('id'));
+    // #enddocregion get-id
+    hero = await (_heroService.getHero(id));
+  }
+// #enddocregion ng-oninit
+
+// #docregion go-back
+  goBack() {
+    window.history.back();
```

Comment:
Shouldn't the `Location` class be used for this (https://github.com/angular/angular/blob/75343eb34007579be9cdc803da834c38e02ae12c/modules/angular2/src/router/location/location.ts#L111) for proper abstraction?
It doesn't matter in this example but I got the impression that Angular tries to promote a style that is compatible with WebWorker and server side rendering.